### PR TITLE
fix(Slider): Update state when properties are updated.

### LINKED ISSUE #1221: PF3: Slider class doesn't update state when properties change and doesn't invoke onSlide when using input form
I've found a couple of limitations when using Slider component (probably it will apply PF3 and PF4).

1) Slider class should update state when properties change
2) When enter a new value from the input form, onSlide should be invoked

These two were kinda blockers for my project but I could workaround it in a pure hack style monkey patching through prototype:

```
Slider.prototype.componentWillReceiveProps = function (nextProps: any) {
  // Update state here like this.state.value = nextProps.value;
}

Slider.prototype.componentDidUpdate = function (prevProps: any, prevState: any) {
  // Catch new state entered from input and inoke your onSlide logic
}
```

### PR DESCRIPTION
**Pull request by lucasponce**
Invoke onSlide when slider-input changes

Fixes #1221

### CODE REVIEW

File: packages/patternfly-3/patternfly-react/src/components/Slider/Slider.js [hunk 1]
```diff
@@ -17,12 +17,19 @@ class Slider extends React.Component {
     };
   }
 
+  componentDidUpdate(prevProps) {
+    if (prevProps.value !== this.props.value) {
+      this.onSlide(this.props.value);
```
[the remark this answers is not in the document]
Comment (lucasponce) [reply packages/patternfly-3/patternfly-react/src/components/Slider/Slider.js]: State needs to be updated when properties change.

[the remark this answers is not in the document]
Comment (jeff-phillips-18) [reply packages/patternfly-3/patternfly-react/src/components/Slider/Slider.js]: exactly

[the remark this answers is not in the document]
Comment (lucasponce) [reply packages/patternfly-3/patternfly-react/src/components/Slider/Slider.js]: I didn't change here the state as it triggers a warning and the CI was blocked and also the onSlide needs to be invoked and it updates the state as side effect.
So I think on this way, it saves one render.

[the remark this answers is not in the document]
Comment (jeff-phillips-18) [reply packages/patternfly-3/patternfly-react/src/components/Slider/Slider.js]: Ah, I see... the onSlide function updates the state 👍 
(it's a bit confusing to have both a this.onSlide and a this.props.onSlide)

**Pull request merged**